import grpc
import metrics_pb2
import metrics_pb2_grpc

def run_client():
    # Conectar ao servidor
    with grpc.insecure_channel('localhost:50051') as channel:
        stub = metrics_pb2_grpc.MonitoringServiceStub(channel)
        
        # 1. Chamada simples
        print("üìä M√©trica √∫nica:")
        response = stub.GetMetric(
            metrics_pb2.MetricRequest(
                hostname="server01",
                metric_type="cpu"
            )
        )
        print(f"  CPU: {response.value:.1f}%")
        
        # 2. Stream de m√©tricas
        print("\nüìà Stream de m√©tricas:")
        request = metrics_pb2.MetricRequest(
            hostname="server01",
            metric_type="memory"
        )
        
        for metric in stub.StreamMetrics(request):
            print(f"  Memory: {metric.value:.2f} GB")
        
        # 3. Health check
        health = stub.CheckHealth(
            metrics_pb2.HealthRequest(service="api")
        )
        print(f"\n‚ù§Ô∏è Health: {health.message}")